### DISCUSSION TITLE
Changing block size of the file system

### DISCUSSION THREAD

**Discussion by bixb922** [micropython/micropython#10754] [ESP32]
I would like to set up the main file system on a ESP32 device with a smaller block size, to be able to store many (200 to 300)  small files (5 kbytes average). Standard block size is 4096 bytes:
```
MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32
Type "help()" for more information.
>>> import os
>>> os.statvfs("/")
(4096, 4096, 512, 510, 510, 0, 0, 0, 0, 255)
```
What I tried is:
```
MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32
Type "help()" for more information.
>>> import esp32
>>> import os
>>> os.umount("/")
>>> bd=esp32.Partition("vfs",512)
>>> os.VfsLfs2.mkfs(bd)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 22] EINVAL
```

Same code with VfsFat instead of VfsLfs2 builds the file system, but on reboot I get:
```
The filesystem appears to be corrupted. If you had important data there, you
may want to make a flash snapshot to try to recover it. Otherwise, perform
factory reprogramming of MicroPython firmware (completely erase flash, followed
by firmware programming).
```
Any other way to achieve smaller block sizes (except rebuilding Micropython)?

**Comment by robert-hh** [comment]
The block size must match the record size of the flash, which is usually 4096 bytes. But as far as I recall tests I made a while ago, you can create more files with LFS2 than number of blocks. LFS is able to share a block between files. That does not apply to FAT.

**Reply by bixb922** [reply]
Thanks for the answer! Testing a bit yields that many small files will fit in one block. The number of files per block can vary between 7 and 70, for files between 128 and <5 bytes. All files larger than 128 bytes seem to get at least one block allocated without block sharing between files. Files between 4097 bytes and 8192 use 2 blocks, no sharing. This is certainly very efficient for small files.
I understand that with LFS2, block size is 4096, there seems to be no easy way to change that.

**Reply by robert-hh** [reply]
Thanks for testing. As far as I recall, I tested that with files of 256 bytes size. 
Note that the 4ß96 bytes values depends on the hardware of the flash chip and not of the files system architecture. The minimum unit for erasing a flash area is determined by the flash architecture, which is 4096 byte for a typical serial flash chip. It can be more, it can be less, but cannot be changed by software settings.
